src/mod_bosh.erl

```diff
@@ -319,27 +319,27 @@ get_session_socket(Sid) ->
     end.
 
 
--spec maybe_start_session(req(), exml:element()) ->
+-spec maybe_start_session(req(), exml:element(), map()) ->
```

Comment:
Why not type-spec them all as `mongoose_listener:options()`? This would give better information about the contents.

Reply:
Because it doesn't have all of the options that belong to the mongoose_listener:options here and then dialyzer would fail :|